whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotObjectSynchronizer.java

```diff
@@ -31,31 +34,33 @@ public class SnapshotObjectSynchronizer {
     private static final int BATCH_SIZE = 10;
 
     private final Dummifier dummifierNrtm;
-    private final NrtmVersionInfoRepository nrtmVersionInfoRepository;
     private final SnapshotObjectRepository snapshotObjectRepository;
+    private final SourceRepository sourceRepository;
     private final WhoisObjectRepository whoisObjectRepository;
 
     SnapshotObjectSynchronizer(
         final Dummifier dummifierNrtm,
-        final NrtmVersionInfoRepository nrtmVersionInfoRepository,
         final SnapshotObjectRepository snapshotObjectRepository,
+        final SourceRepository sourceRepository,
         final WhoisObjectRepository whoisObjectRepository
     ) {
         this.dummifierNrtm = dummifierNrtm;
-        this.nrtmVersionInfoRepository = nrtmVersionInfoRepository;
         this.snapshotObjectRepository = snapshotObjectRepository;
+        this.sourceRepository = sourceRepository;
         this.whoisObjectRepository = whoisObjectRepository;
     }
 
-    NrtmVersionInfo initializeSnapshotObjects(final NrtmSource source) {
-        final String method = "initializeSnapshotObjects";
+    InitialSnapshotState initializeSnapshotObjects() {
+        LOGGER.info("initializeSnapshotObjects entered");
+        final Map<CIString, NrtmSourceModel> sourceMap = new HashMap<>();
+        for (final NrtmSource source : NrtmSourceHolder.getAllSources()) {
```

Comment:
instead of doing the loop on this list twice (NrtmSourceHolder.getAllSource())
you could do it once in the SnapshotFileGenerator and pass the Map<CIString, NrtmSourceModel> as a parameter. I think that here you could avoid one loop

Reply:
they're used in slightly different ways, so i'd still need to loop or refactor the code to make both uses compatible. that's ok except the code needs changes for the next story so i'd prefer to keep it like this for now. there's no I/O or memory implications so it will be really fast anyway.